### PR TITLE
Fix memory leak of PyBytes -> Vec conversion

### LINKED ISSUE #570: Memory leak on extracting Vec<u8>
Extracting a Vec of u8s from pyobject leaks memory.

### 🌍 Environment

 - Your operating system and version:

Linux 3bda9a19dfcb 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

 - Your python version:

Python 3.6.8

 - How did you install python (e.g. apt or pyenv)? Did you use a virtualenv?:

System install inside docker container, installed via apt

 - Your rust version (`rustc --version`):

rustc 1.37.0-nightly (5eeb567a2 2019-06-06)

 - Are you using the latest pyo3 version? Have you tried using latest master (replace `version = "0.x.y"` with `git = "https://github.com/PyO3/pyo3")?` 

pyo3 = { version = "0.7.0", features = ["extension-module"] }

### 💥 Reproducing
```rust
use pyo3::prelude::*;

#[pyclass]
struct MyClass {}

#[pymethods]
impl MyClass {
    #[new]
    pub fn __new__(obj: &PyRawObject, _py: Python) {
        obj.init({ MyClass {} });
    }

    pub fn leaks(&mut self, py: Python, py_byte_string: PyObject) -> PyResult<()> {
        let byte_vec: Vec<u8> = py_byte_string.extract(py).unwrap();
        println!("{:?}", byte_vec.len());
        Ok(())
    }

    pub fn doesnt_leak(&mut self, py: Python, py_byte_string: PyObject) -> PyResult<()> {
        let byte_vec: String = py_byte_string.extract(py).unwrap();
        println!("{:?}", byte_vec.len());
        Ok(())
    }
}

#[pymodule]
pub fn my_module(_py: Python, m: &PyModule) -> PyResult<()> {
    m.add_class::<MyClass>()?;
    Ok(())
}
```

Running in python 
```python
import os
import psutil
import pyo3_memory_leak

process = psutil.Process(os.getpid())
initial_mem = process.memory_info().rss

my_instance = pyo3_memory_leak.my_module.MyClass()

for i in range(10**6):
    my_instance.doesnt_leak("Hello world")

middle_mem_1 = process.memory_info().rss

for i in range(10**6):
    my_instance.doesnt_leak("Hello world")

middle_mem_2 = process.memory_info().rss

for i in range(10**6):
    my_instance.leaks(b"Hello world")

final_mem_1 = process.memory_info().rss

for i in range(10**6):
    my_instance.leaks(b"Hello world")

final_mem_2 = process.memory_info().rss


print("Inital mem", initial_mem)
print("Middle mem 1", middle_mem_1)
print("Middle mem 2", middle_mem_2)
print("Final mem 1", final_mem_1)
print("Final mem 2", final_mem_2)
```

Produces output

```
Inital mem 	 19783680
Middle mem 1 	 19783680
Middle mem 2 	 19783680
Final mem 1 	 115871744
Final mem 2 	 211841024
```

Digging into code suspect something in this unsafe block

https://github.com/PyO3/pyo3/blob/98b707918315e2db3a90ea953177b20a5cdeff49/src/buffer.rs#L580

### PR DESCRIPTION
**Pull request by kngwyu**
Resolves #570.
For now, I only added a test.

### CODE REVIEW

File: src/buffer.rs [hunk 1]
```diff
@@ -582,7 +582,6 @@ impl PyBuffer {
             let ptr = &*self.0 as *const ffi::Py_buffer as *mut ffi::Py_buffer;
             ffi::PyBuffer_Release(ptr)
         };
-        mem::forget(self);
```
Comment (kngwyu) [inline src/buffer.rs]: So what on earth this `forget` mean?
Since the pointer that `self` has is already released, there's no reason for keeping it.

**Comment by kngwyu** [comment]
Could I include PyPy in `allow_failures` section in travis?
This segfault is exactly the same as #589 and not related to this PR.

**Comment by konstin** [comment]
I'd prefer skipping only the failing check with some cfg.

**Comment by kngwyu** [comment]
@konstin 
OK, I added `pytest.skipif` decorator for the test.

File: src/buffer.rs (hunk 1 above)
Comment (konstin) [reply src/buffer.rs]: It means that `drop` shouldn't be called

**Pull request merged**